service/topology_coordinator.cc

```diff
@@ -740,6 +755,56 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
         }
     }
 
+    // If a node crashes after initiating gossip and before joining group0, it becomes orphan and 
+    // remains in gossip. This backgound fiber periodically checks for such nodes and purges those 
+    // entries from gossiper by commiting the node as left in raft.
+    future<> gossiper_orphan_remover_fiber() {
```

Comment:
Do we need separate fiber? May be the check can be added to `gossiper::do_status_check` which already runs periodically. It does not run currently in raft topology mode, but this is because it has nothing to do.

Reply:
> Correct me if I am wrong. 

You are very right. I do not propose to just re-enable the status check as is, but run your logic from this patch in the status check in case raft topology is enabled. IOW do not remove from gossiper if the node is in the raft topology.

Reply:
Okay, let me describe a very rare case If I use gossiper based mechanism for purging.

Let the sleepy node be A. It just sleeps for 60s, doesn't crash. As you described, the nodes will notice this node's timestamp to be older than 60s and will decide to purge it. Now, the node is still alive and has sent the join_node_request to the topology coordinator at 60th second. Before commiting the new node as topology mutation, the TC confirms whether ip is available in address_map via `wait_for_ip`. If these checks are passed, it commits the mutation and adds the new node into topology. Since our node A's entry is present in gossiper, TC will commit the topology change. But what if the gossiper purges the entries before commit goes through and after the `wait_for_ip` validation? Gossiper will check the topology object and find the sleepy node to be absent and will purge it.

Order of events: 
1. wait_for_ip check
2. Gossiper purge
3. topology change commit

In this rare case, the node will join cluster and it will be unreachable. Nodes like these are threat to achieving majority in raft operations.

Now why banning is necessary is a separate topic, we just saw that we need to use raft for this purging business, and moving the node to left state give us this mechanism for free and effective manner. Since we don't have tombstone based removal, banning ensures the removal of node in gossiper.